### PR TITLE
Improve codes

### PR DESCRIPTION
**Pull request by minggo**
## Summary by Sourcery

Improve the user registration flow by consolidating and enforcing uniqueness on username and email, introducing unique DB indexes, and adding robust error handling for both duplicate key and email delivery failures.

Enhancements:
- Consolidate username/email existence check into a single query and unified error response
- Wrap user and code insertion in a try/catch to handle DuplicateKeyError gracefully
- Add creation of unique indexes on the users collection at application startup
- Add exception handling in the email utility to map SMTP errors to HTTPExceptions

Tests:
- Update duplicate-username test to expect combined error message
- Add end-to-end test for duplicate-email registration path and verify single user and code entries

Chores:
- Introduce new module to initialize database indexes

### CODE REVIEW

File: server/app/utils/email.py [hunk 1]
```diff
@@ -13,7 +15,16 @@ def send_verification_email(to_email: str, code: str) -> None:
...
+    try:
+        with smtplib.SMTP(settings.SMTP_HOST, settings.SMTP_PORT) as smtp:
+            smtp.starttls()
+            smtp.login(settings.SMTP_USER, settings.SMTP_PASSWORD)
+            smtp.send_message(msg)
+    except SMTPAuthenticationError as e:
+        raise HTTPException(status_code=500, detail="SMTP authentication failed")
+    except SMTPConnectError as e:
+        raise HTTPException(status_code=500, detail="Failed to connect to SMTP server")
+    except SMTPException as e:
```
[the remark this answers is not in the document]
Comment (minggo) [reply server/app/utils/email.py]: fixed

File: server/app/api/auth.py [hunk 2]
```diff
@@ -38,32 +39,33 @@ async def register(
...
+        expires = datetime.utcnow() + timedelta(minutes=settings.VERIFICATION_CODE_EXPIRE_MINUTES)
+        await db[COLLECTION_CODES].insert_one({
+            "user_id": user_id,
+            "email": req.email,
+            "code": code,
+            "expires_at": expires,
+            "created_at": datetime.utcnow(),
+        })
+    except DuplicateKeyError:
+        raise HTTPException(status_code=400, detail="Username or email already exists")
```
[the remark this answers is not in the document]
Comment (minggo) [reply server/app/api/auth.py]: fixed
